src/indexer/segment_writer.rs

```diff
@@ -900,4 +911,42 @@ mod tests {
         postings.positions(&mut positions);
         assert_eq!(positions, &[4]); //< as opposed to 3 if we had a position length of 1.
     }
+
+    // ISSUE-#2078 - writing and searching shall throw error when the field tokenizer is missing
+    #[test]
+    fn test_show_error_when_tokenizer_not_registered() {
+        let text_field_indexing = TextFieldIndexing::default()
+            .set_tokenizer("custom_en")
+            .set_index_option(IndexRecordOption::WithFreqsAndPositions);
+        let text_options = TextOptions::default()
+            .set_indexing_options(text_field_indexing)
+            .set_stored();
+        let custom_en_tokenizer = TextAnalyzer::builder(WhitespaceTokenizer::default())
+            .filter(RemoveLongFilter::limit(40))
+            .filter(Stemmer::new(Language::English))
+            .build();
+        let mut schema_builder = Schema::builder();
+        schema_builder.add_text_field("title", text_options);
+        let schema = schema_builder.build();
+        let tempdir = TempDir::new().unwrap();
+        let tempdir_path = PathBuf::from(tempdir.path());
+        let index = Index::create_in_dir(&tempdir_path, schema).unwrap();
+        index
+            .tokenizers()
+            .register("custom_en", custom_en_tokenizer);
```

Comment:
I think that can be removed

Reply:
I guess we can remove the tokenizer registration but we may need the index creation else `open_in_dir` will fail